### PR TITLE
[TE] Severity based alerter

### PR DESCRIPTION
**Pull request by jihaozh**
## Description
This PR adds the severity based alerting. It sends out anomalies to different email/Jira channels based on its anomaly severity. It also supports re-sending the alert if a given anomaly's severity has changed. This is achieved by adding a new lookup table for recording the relationship between an anomaly and subscription groups.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionPipelineTaskRunner.java [hunk 1]
```diff
@@ -167,6 +175,14 @@ public DetectionPipelineTaskRunner(DetectionConfigManager detectionDAO, MergedAn
       }
       this.detectionDAO.update(config);
 
+      // re-notify the anomalies if any
+      for (MergedAnomalyResultDTO anomaly : result.getAnomalies()) {
+        // if an anomaly should be re-notified, update the notification lookup table in the database
+        if (anomaly.shouldRenotify()) {
```
Comment (vincentchenjl) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionPipelineTaskRunner.java:181]: Do we expect that new anomalies also have `renotify` flag as true? Or this should only apply for existing anomalies?

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/constant/AnomalySeverity.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.apache.pinot.thirdeye.constant;
+
+public enum AnomalySeverity {
+  LOW("low"), MODERATE("moderate"), HIGH("high"), CRITICAL("critical"), DEFAULT("default");
```
Comment (vincentchenjl) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/constant/AnomalySeverity.java:24]: There's a conflict here between this PR and mine. The labeler in the other PR relies on the order of the severity.

**Review by vincentchenjl — COMMENTED** [review]
Overall LGTM! We need to resolve some conflicts offline

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionResource.java [hunk 3]
```diff
@@ -570,4 +574,19 @@ public Response getDetectionHealth(@PathParam("id") @ApiParam("detection config
...
+  @Path(value = "/re-notify")
+  @ApiOperation("Resend the notification for the anomalies to the subscribed notification groups, if the subscription group supports re-notify")
+  public Response alert(@QueryParam("id") List<Long> anomalyIds) {
+    List<MergedAnomalyResultDTO> anomalies = this.anomalyDAO.findByIds(anomalyIds);
+    for (MergedAnomalyResultDTO anomaly : anomalies) {
+      AnomalySubscriptionGroupNotificationDTO anomalySubscriptionGroupNotificationDTO =
+          new AnomalySubscriptionGroupNotificationDTO();
+      anomalySubscriptionGroupNotificationDTO.setAnomalyId(anomaly.getId());
+      anomalySubscriptionGroupNotificationDTO.setDetectionConfigId(anomaly.getDetectionConfigId());
+      anomalySubscriptionGroupNotificationManager.save(anomalySubscriptionGroupNotificationDTO);
```
[the remark this answers is not in the document]
Comment (jihaozh) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionResource.java]: good catch. updated!

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionPipelineTaskRunner.java (hunk 1 above)
Comment (jihaozh) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionPipelineTaskRunner.java:181]: no. the renotify flag only applies to exist anomalies.

**Review by akshayrai — APPROVED** [review]

**Pull request merged**